Siliconlabs HCI driver improvements

petrip-slab opened the pull request:
- Move Siliconlabs specific HCI Kconfigs to own file.
- use proper macros for enabling features
- add support for controller privacy to enable PTS test runs

petrip-slab committed:
silabs: drivers: bluetooth: Move configurations to Kconfig

Several driver setting were as defines in driver file which made hard to
change by application programmers. Fixed by adding
silicon labs specific Kconfig file for bluetooth driver configurations.

Also some cleaning for handling Kconfig conditional to conform
preferred style.

Signed-off-by: Petri Pitkanen <petri.pitkanen@silabs.com>

petrip-slab committed:
silabs: drivers: bluetooth: Add support for controller privacy

Added support for contoroller privacy support.

Signed-off-by: Petri Pitkanen <petri.pitkanen@silabs.com>

jerome-pouiller commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -241,6 +241,7 @@ config BT_AIROC
 source "drivers/bluetooth/hci/Kconfig.esp32"
 source "drivers/bluetooth/hci/Kconfig.infineon"
 source "drivers/bluetooth/hci/Kconfig.nxp"
+source "drivers/bluetooth/hci/Kconfig.siliconlabs"
```
Usualy, we use "silabs" rather than "siliconlabs".

jerome-pouiller commented on drivers/bluetooth/hci/Kconfig.siliconlabs:
```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2024 SiliconLabs <www.silabs.com/>
```
We usually use: `Copyright (c) 2025 Silicon Laboratories Inc.`

jerome-pouiller commented on drivers/bluetooth/hci/Kconfig.siliconlabs:
```diff
@@ -0,0 +1,46 @@
+
+config BT_SILABS_EFR32_BUFFER_MEMORY
+	int "Silicon Labs Bluetooth Library memory buffer size"
+	depends on BT_SILABS_EFR32
```
I suggest to place all these options under `menuconfig BT_SILABS_EFR32` rather than repeating `depends on BT_SILABS_EFR32`

jerome-pouiller commented on drivers/bluetooth/hci/Kconfig.siliconlabs:
```diff
@@ -0,0 +1,46 @@
+	  Bluetooth Library.
+
+config BT_SILABS_EFR32_USER_ADVERTISERS
+	int "Silicon Labs Bluetooth Library user advertisement sets"
```
I believe you won't have to repeat `Silicon Labs Bluetooth Library` if you use `menuconfig`.

jerome-pouiller commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
Typo in the commit log: "contoroller"

jerome-pouiller commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
```diff
@@ -21,16 +21,11 @@ struct hci_data {
-static K_KERNEL_STACK_DEFINE(slz_ll_stack, SL_BT_SILABS_LL_STACK_SIZE);
+#if !defined(CONFIG_BT_MAX_CONN)
+#define CONFIG_BT_MAX_CONN 0
+#endif
```
Is it useful?

petrip-slab replied:
I was recommended to us IS_ENABLED  and the init command that is behind this IS_ENABLED(connections) refers
to the BT_MAX_CONN and then the file wont compile without connections

The other solution would be jus "#if defined" kinda solutin and not use IS_DEFINED but then syntax of thing wont get checked on every compilation.

jhedberg replied:
I don't think it's a good idea to redefine Kconfig preprocessor macros. One should be able to trust that whatever the value the macro has it's coming from Kconfig files. I'd use a local macro here instead, e.g. just `MAX_CONN`:
```c
#if defined(CONFIG_BT_MAX_CONN)
#define MAX_CONN CONFIG_BT_MAX_CONN
#else
#define MAX_CONN 0
#endif
```

That said, I think it's fair to question whether the `CONFIG_BT_MAX_CONN` and `CONFIG_BT_CTLR_RL_SIZE` should have their current dependencies, i.e. why couldn't they just have the `BT` and `BT_CTLR` dependencies respectively and then default to `0` when `BT_CONN` or `CONFIG_BT_RL` (or whatever its feature option is called) are not enabled.

jerome-pouiller replied:
Maybe we could reuse the `IF_EXISTS()` macro introduced by @asmellby:
https://github.com/zephyrproject-rtos/zephyr/pull/83805/commits/17791ec0b028646bc3d820fd486ab7fa482405c3#diff-3529e58b55964920621c09466f4a305145918ca46c9fa12e7341ba019d36b485R14-R15

petrip-slab replied:
I tried that and ended up having 0 connections. May it due it being very flag oriented? 
Atttempt to read to chain of #defines down to bottom gave nothing but headache.

jerome-pouiller replied:
Indeed, `IF_EXISTS()` does not work as expected. Sorry for the disturbance.

jerome-pouiller commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
```diff
@@ -24,6 +24,9 @@ struct hci_data {
 #endif
+#if !defined(CONFIG_BT_CTLR_RL_SIZE)
+#define CONFIG_BT_CTLR_RL_SIZE 0
+#endif
```
Ditto.

petrip-slab committed:
silabs: drivers: bluetooth: Fix temporarily observer configuration

Link layer needs some fixing before hci driver can be minimally configured
and hence must revert conditional inclusion of core functionalities untill
new link layer is available.

Signed-off-by: Petri Pitkanen <petri.pitkanen@silabs.com>

petrip-slab commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -241,6 +241,7 @@ config BT_AIROC
 source "drivers/bluetooth/hci/Kconfig.esp32"
 source "drivers/bluetooth/hci/Kconfig.infineon"
 source "drivers/bluetooth/hci/Kconfig.nxp"
+source "drivers/bluetooth/hci/Kconfig.silabs"
```
Yes it is more logical. other "source" directives just bundled together so I assumed it is a practice. now fixec

petrip-slab commented:
added one more  commit to workaround the problems observed in "observer" sample

jerome-pouiller commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
```diff
@@ -26,15 +26,15 @@ struct hci_data {
 #define MAX_CONN 0
 #endif
-<<<<<<< HEAD
-=======
```
Place this change the previous commit.

jhedberg commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
```diff
@@ -21,16 +21,22 @@ struct hci_data {
+#define MAX_CONN 0
+#endif
+<<<<<<< HEAD
+=======
```
I missed this since I only looked at the final diff. Please get rid of this kind of incorrect stuff from the intermediate commits.

jhedberg commented on drivers/bluetooth/hci/hci_silabs_efr32.c:
```diff
@@ -21,16 +21,22 @@ struct hci_data {
+#else
+#define CTLR_RL_SIZE 0
+#endif
+>>>>>>> dd0edd47462 (fixup! silabs: drivers: bluetooth: Move configurations to Kconfig)
```
Same here
